src/test/groovy/net/rubygrapefruit/platform/file/AbstractFileEventFunctionsTest.groovy

```diff
@@ -422,26 +460,57 @@ abstract class AbstractFileEventFunctionsTest extends Specification {
         }
     }
 
-    protected String shorten(FileWatchEvent event) {
-        return event.type.name() + " " + shorten(String.valueOf(event.path))
+    protected String format(FileWatchEvent event) {
+        String shortened = null
+        event.handleEvent(new FileWatchEvent.Handler() {
+            @Override
+            void handleChangeEvent(ChangeType type, String absolutePath) {
```

Comment:
One option to make this nicer is to parameterize the `Handler` with a generic variable `T`, and then have each of the `handle*` method return `T`. That way, `handleEvent` can also return `T`.

If we only need this for tests though, then the interface can also stay as is.